core/src/main/resources/hudson/tasks/Messages.properties

```diff
@@ -81,3 +83,5 @@ Maven.NotMavenDirectory={0} doesn\u2019t look like a Maven directory
 Maven.NoExecutable=Couldn\u2019t find any executable in {0}
 
 Shell.DisplayName=Execute shell
+Shell.invalid_exit_code_range=Invalid exit code value: {0}. Check help section
+Shell.invalid_exit_code_zero=Exit code zero is not allowed to make build unstable
```

Comment:
Form validation does not prevent users from saving the form. I expect the '0' value to actually be saved. So, *strongly discouraged*, with explanation that it's never going to be successful perhaps?